### PR TITLE
Dot product on ARM

### PR DESCRIPTION
**Pull request by aneshlya**
Fixes https://github.com/ispc/ispc/issues/2951

### CODE REVIEW

File: tests/lit-tests/aarch64_dotprod.ispc [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Test checks emitted code for ARM (aarch64) dot product instructions.
```
Comment (nurmukhametov) [inline tests/lit-tests/aarch64_dotprod.ispc:1]: Do we need a test to ensure that udot/sdot instructions are not generated on CPUs that do not support them?

**Review by nurmukhametov — APPROVED** [review]
LGTM!

File: stdlib/include/builtins.isph [hunk 2]
```diff
@@ -309,11 +309,15 @@ EXT inline READNONE varying int32 __insert_int32(varying int32, uniform int32, u
 EXT inline READNONE varying int64 __insert_int64(varying int64, uniform int32, uniform int64);
 EXT inline READNONE varying int8 __insert_int8(varying int8, uniform int32, uniform int8);
 
-// dor product
+// dot product
 EXT READNONE varying int32 __dot2add_i16packed_sat(varying uint32, varying uint32, varying int32);
 EXT READNONE varying int32 __dot2add_i16packed(varying uint32, varying uint32, varying int32);
 EXT READNONE varying int32 __dot4add_u8i8packed_sat(varying uint32, varying uint32, varying int32);
 EXT READNONE varying int32 __dot4add_u8i8packed(varying uint32, varying uint32, varying int32);
+EXT READNONE varying uint32 __dot4add_u8u8packed_sat(varying uint32, varying uint32, varying uint32);
```
Comment (dbabokin) [inline stdlib/include/builtins.isph:317]: 32 bit saturation doesn't make sense for 8 bit data types. 8*8 gives 16 bit result. So 16 bit saturation does make sense, but not 32 bit.

Comment (dbabokin) [reply stdlib/include/builtins.isph:317]: I haven't realized that it accumulates with 32 bit accumulator. So saturation does make sense, but only on the last step `acc + tmp1234`

File: tests/lit-tests/aarch64_dotprod.ispc (hunk 1 above)
Comment (aneshlya) [reply tests/lit-tests/aarch64_dotprod.ispc:1]: Added

File: stdlib/include/builtins.isph (hunk 2 above)
Comment (aneshlya) [reply stdlib/include/builtins.isph:317]: Removed unnecessary saturations

**Review by dbabokin — APPROVED** [review]

**Pull request merged**